### PR TITLE
Implement render of `TransferAsset`.

### PR DESCRIPTION
**Pull request by Unengine**
### Description
SSIA

### Required Reviewers
@planetarium/9c-dev 

### Screenshots
image
image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1141562434100787/1200503063713152) by [Unito](https://www.unito.io)

### CODE REVIEW

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 1]
```diff
@@ -267,6 +267,14 @@ private void ClaimMonsterCollectionReward()
                 .Subscribe(ResponseClaimMonsterCollectionReward).AddTo(_disposables);
         }
 
+        private void TransferAsset()
+        {
+            _renderer.EveryRender<TransferAsset>()
+                .Where(ValidateEvaluationForCurrentAgent)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:273]:
```suggestion
                .Where(HasUpdatedAssetsForCurrentAgent)
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs [hunk 2]
```diff
@@ -146,6 +147,15 @@ private void ClaimMonsterCollectionReward()
                 .AddTo(_disposables);
         }
 
+        private void TransferAsset()
+        {
+            _renderer.EveryUnrender<TransferAsset>()
+                .Where(ValidateEvaluationForCurrentAgent)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:153]:
```suggestion
                .Where(HasUpdatedAssetsForCurrentAgent)
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 3]
```diff
@@ -999,6 +1007,28 @@ private void ResponseClaimMonsterCollectionReward(ActionBase.ActionEvaluation<Cl
...
+        {
+            if (!(eval.Exception is null))
+            {
+                return;
+            }
+
+            var recipientAddress = eval.Action.Recipient;
+            var currentAgentAddress = States.Instance.AgentState.address;
+
+            if (recipientAddress == currentAgentAddress)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:1020]: We need to consider senderAddress equals currentAgentAddress either.

File: nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs [hunk 4]
```diff
@@ -368,6 +378,22 @@ private void ResponseClaimMonsterCollectionReward(ActionBase.ActionEvaluation<Cl
...
+        {
+            if (!(eval.Exception is null))
+            {
+                return;
+            }
+
+            var recipientAddress = eval.Action.Recipient;
+            var currentAgentAddress = States.Instance.AgentState.address;
+
+            if (recipientAddress == currentAgentAddress)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:391]: We need to consider senderAddress equals currentAgentAddress either.

**Review by boscohyun — CHANGES_REQUESTED** [review]

**Review by boscohyun — APPROVED** [review]

File: nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs [hunk 5]
```diff
@@ -385,10 +385,12 @@ private void ResponseTransferAsset(ActionBase.ActionEvaluation<TransferAsset> ev
                 return;
             }
 
+            var senderAddress = eval.Action.Sender;
             var recipientAddress = eval.Action.Recipient;
             var currentAgentAddress = States.Instance.AgentState.address;
 
-            if (recipientAddress == currentAgentAddress)
+            if (recipientAddress == currentAgentAddress ||
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:392]: I think `recipientAddress == currentAgentAddress &&`
|| -> &&

Comment (Unengine) [reply nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:392]: This should be rendered both sender and recipient, so I used `or` condition.

Comment (hskim881028) [reply nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:392]: ahh! i understand

**Review by hskim881028 — APPROVED** [review]

**Pull request merged**